### PR TITLE
[5787] - automate sign off period content switching

### PR DESCRIPTION
**Pull request by d-a-v-e**
### Context

Some of the guidance and reports content only makes sense inside the census or performance profiles sign off periods.

### Changes proposed in this pull request

* Adds a service to determine what period we are in
* uses this to switch content in the guidance index page (this is actually under `app/views/guidance/show.html.erb`)
* the content of the reports index page (`app/views/reports/index.html.erb`)

### Guidance to review

* Run locally and have the `app/services/sign_off_period_service.rb` return the desired period for checking each of the three states (`:census_period`, `:performance_period`, `:outside_period`)
* look at the [trello ticket](https://trello.com/c/ERWgb7NB/5787-switch-between-alternative-versions-of-reports-and-guidance-for-census-and-performance-profiles-outside-sign-off-periods) to see what content is effected.

### CODE REVIEW

**Comment by d-a-v-e** [comment]
[Switch between alternative versions of reports and guidance for census and performance profiles outside sign off periods](https://trello.com/c/ERWgb7NB/5787-switch-between-alternative-versions-of-reports-and-guidance-for-census-and-performance-profiles-outside-sign-off-periods)

**Comment by carlosmartinez** [comment]
@d-a-v-e should we get this on a review app for the sake of product review?

**Comment by d-a-v-e** [comment]
> @d-a-v-e should we get this on a review app for the sake of product review?

@carlosmartinez It will be awkward triggering the different states - I was planning to go over it locally with Sarah tomorrow morning.

**Comment by carlosmartinez** [comment]
> > @d-a-v-e should we get this on a review app for the sake of product review?
> 
> @carlosmartinez It will be awkward triggering the different states - I was planning to go over it locally with Sarah tomorrow morning.

ok that's reasonable. we could consider building a period switcher into the support console for review apps and QA.

File: app/services/sign_off_period_service.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class SignOffPeriodService
```
Comment (carlosmartinez) [inline app/services/sign_off_period_service.rb:3]: suggest name `DetermineSignOffPeriod`

File: app/views/reports/_census_period.html.erb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<ul class="govuk-list govuk-list--spaced">
+  <li>
+    <p>
+      <%= govuk_link_to "New trainees for the #{@current_academic_cycle_label} academic year", itt_new_starter_data_sign_off_reports_path, class: "new-trainee-data" %>
+      - for census sign off
+    </p>
+  </li>
+  <li>
+    <p>
+      <%= "Trainees who studied in the #{@previous_academic_cycle_label} academic year - will be available for performance profiles sign off" %>
```
Comment (carlosmartinez) [inline app/views/reports/_census_period.html.erb:10]:
```suggestion
      Trainees who studied in the <%= @previous_academic_cycle_label} %> academic year - will be available for performance profiles sign off
```

File: app/views/reports/_outside_period.html.erb [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <p class="govuk-body">No reports are currently available.</p>
 <p class="govuk-body">When they’re available, you’ll be able to download reports showing:</p>
 <ul class="govuk-list govuk-list--bullet">
-  <li>new trainees for the 2022 to 2023 academic year - for census sign off</li>
-  <li>trainees who studied in the 2021 to 2022 academic year - for performance profiles sign off</li>
+  <li> <%= "new trainees for the #{@current_academic_cycle_label} academic year - for census sign off" %> </li>
```
Comment (carlosmartinez) [inline app/views/reports/_outside_period.html.erb:4]:
```suggestion
  <li> new trainees for the <%= @current_academic_cycle_label %> academic year - for census sign off</li>
```

File: app/views/reports/_performance_period.html.erb [hunk 4]
```diff
@@ -1,8 +1,7 @@
 <ul class="govuk-list govuk-list--spaced">
   <li>
     <p>
-      <%= govuk_link_to "New trainees for the #{@current_academic_cycle_label} academic year", itt_new_starter_data_sign_off_reports_path, class: "new-trainee-data" %>
-      - for census sign off
+      <%= "New trainees for the #{@current_academic_cycle_label} academic year - will be available for census sign off" %>
```
Comment (carlosmartinez) [inline app/views/reports/_performance_period.html.erb:4]:
```suggestion
      New trainees for the <%= @current_academic_cycle_label %> academic year - will be available for census sign off
```

**Comment by carlosmartinez** [comment]
looks good, just a few small comments/suggestions

File: app/services/sign_off_period_service.rb (hunk 1 above)
Comment (d-a-v-e) [reply app/services/sign_off_period_service.rb:3]: thanks, that reads better 👍

**Review by carlosmartinez — APPROVED** [review]

**Pull request merged**